src/lib.rs

```diff
@@ -1914,6 +1915,50 @@ pub trait Itertools: Iterator {
     }
 
     // non-adaptor methods
+    /// Advances the iterator and returns the next items grouped in an array of
+    /// a specific size.
+    ///
+    /// If there are enough elements to be grouped in an array, then the array
+    /// is returned inside `Some`, otherwise `None` is returned.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let mut iter = 1..5;
+    ///
+    /// assert_eq!(Some([1, 2]), iter.next_array());
+    /// ```
+    fn next_array<T, const N: usize>(&mut self) -> Option<[T; N]>
```

Comment:
`T` is the same as `Self::Item`, _we don't need a generic here!_
Same for `collect_array` and the private `next_array::next_array`.

@jswrenn I don't feel competent enough to review unsafe parts, I'll let you judge.